packages/service-library/src/data-providers/document-provider.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { client, CosmosOperationResponse } from 'azure-services';
+import { injectable } from 'inversify';
+
+@injectable()
+export abstract class DocumentProvider {
+    protected async executeQueryWithContinuationToken<T>(execute: (token?: string) => Promise<CosmosOperationResponse<T[]>>): Promise<T[]> {
```

Comment:
this is duplicate code from PageDocumentProvider Can we move it under /azure-services/CosmosClientWrapper/ package to expose and refactor out duplicates